src/matcher/index.ts

```diff
@@ -71,11 +72,37 @@ export function createRouterMatcher(
     return matcherMap.get(name)
   }
 
+  function routeComponentsOptionGuard(
+    recordComponents?: Record<string, RawRouteComponent>
+  ): never | void {
+    // Guard for record.components. See Issue #1260
+    // Recieved components: Home
+    // This should be components: { default: Home, others } or component: Home
+    for (const name in recordComponents) {
+      const rawComponent = recordComponents[name]
+      // property components should be object
+      if (
+        !rawComponent ||
+        (typeof rawComponent !== 'object' && typeof rawComponent !== 'function')
+      ) {
+        warn(
+          `Components property in record is not valid. ` +
+            `Property "components" should be component: MyView or ` +
+            `components: { default: MyView }`
+        )
+        // same with navigation guards
+        throw new Error(`Invalid route record "components" property.`)
+      }
+    }
```

Comment:
let's just use the isRouteComponent (or similar name) to check if recordComponents is a component, and if that's the case, return `{ default: components }` to auto fix the issue during dev

Reply:
@posva Thanks for review. One Quick question about mentioned `isRouteComponent` 

Using isRouteComponent should be like below?

```javascript
  function routeComponentsOptionGuard() {
    for (const name in recordComponents) {
      const rawComponent = recordComponents[name]
      if (isRouteComponent(rawComponent) // ⭐️ ) {
                // ...
      }
    }
    return recordComponents;
  }

function isRouteComponent(
  component: RawRouteComponent
): component is RouteComponent {
  return (
    typeof component === 'object' ||
    'displayName' in component ||
    'props' in component ||
    '__vccOpts' in component
  )
}
```

How can I guard precisely the difference between `single component = {}` and `{ default: single component = {} }`?

Reply:
I pushed what I worked. (please refer last commit wip: tag attached)